source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -663,6 +659,13 @@ Before a given `Connection <#connection>`_ is returned from checkOut, it must be
     else:
         decrement availableConnectionCount
     set connection state to "in use"
+
+    # If there is no background thread, the pool MUST ensure that
+    # there are at least minPoolSize total connections.
+    do asynchronously:
+      while totalConnectionCount < minPoolSize:
+        populate the pool with a connection
+
```

Comment:
1. Moved the part of the `checkOut` logic that is concerned with populating the pool at the end of the `checkOut` logic because the previous position was weird: between creating a new connection that is about to be checked out and establishing it.
2. Replaced `This MUST be done in a non-blocking manner` with the `do asynchronously` block, because it's the only interpretation of "non-blocking manner" I can see here, and I find it clearer. Expressing the requirement as pseudocode is also more explicit as opposed to being expressed in a comment.